### PR TITLE
materialItem: validate that quantity is greater than 0

### LINKED ISSUE #5077: Disallow negative numbers for MaterialItem quantity
> - Should we disallow negative numbers?
Maybe on the validation layer

_Originally posted by @manuelmeister in https://github.com/ecamp/ecamp3/issues/5017#issuecomment-2067923599_

Should be validated on backend and frontend.

### PR DESCRIPTION
**Pull request by BacLuc**
closes #5077

### CODE REVIEW

**Review by pmattmann — APPROVED** [review]

File: api/src/Entity/MaterialItem.php [hunk 1]
```diff
@@ -110,6 +110,7 @@ class MaterialItem extends BaseEntity implements BelongsToCampInterface, CopyFro
      * The number of items or the amount in the unit of items that are required.
      */
     #[ApiProperty(example: 1.5)]
+    #[Assert\GreaterThan(0)]
```
Comment (simfeld) [inline api/src/Entity/MaterialItem.php:113]: are there any implications for existing (now invalid) data?

File: frontend/src/locales/de.json [hunk 2]
```diff
@@ -512,6 +512,7 @@
       "copied": "{source} kopiert"
     },
     "validation": {
+      "greaterThan": "{_field_} muss grösser sein als {min}",
```
Comment (simfeld) [inline frontend/src/locales/de.json:515]: Looks not so great in the UI, can we improve this?
grafik

**Review by usu — APPROVED** [review]

File: api/src/Entity/MaterialItem.php (hunk 1 above)
Comment (BacLuc) [reply api/src/Entity/MaterialItem.php:113]: yes, existing values with invalid content cannot be saved anymore until you make a valid value out of it

File: frontend/src/locales/de.json (hunk 2 above)
Comment (BacLuc) [reply frontend/src/locales/de.json:515]: This is a separate bug in firefox for all validations: 
image
-> new issue

Comment (BacLuc) [reply frontend/src/locales/de.json:515]: - https://github.com/ecamp/ecamp3/issues/5691

**Pull request merged**